src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -104,8 +104,11 @@ class CredentialType(Enum):  # pylint: disable=too-few-public-methods
 
 
 class Profile(object):
-    def __init__(self, cli_ctx, storage=None, auth_ctx_factory=None, use_global_creds_cache=True):
-        self.cli_ctx = cli_ctx
+
+    def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=True, cli_ctx=None):
```

Comment:
The kwarg order has to be preserved here because `azure.common` relies on storage being the first in the list.  This is the reason for all the other changes in this PR.  When azure.common is updated to specify the kwarg instead of relying on positional arguments, these changes can be reverted (or left in place since positional arguments seem to be the source of all tears...)